server/routes/comments.js

```diff
@@ -1,80 +1,95 @@
-    try {
-        let commentByUserID = await db.any(`SELECT * FROM comments WHERE users_id = '${req.params.users_id}' `)
-        res.json({
-            payload: commentByUserID,
-            message:"got all comments by users id"
-        })
-    }catch (err){
-        console.log(err)
-        res.json({
-            error:err
-        })
-    }
-})
+// router.get('/:users_id', async (req, res) => {
+//     try {
+//         let commentByUserID = await db.any(`SELECT * FROM comments WHERE users_id = '${req.params.users_id}' `)
+//         res.json({
+//             payload: commentByUserID,
+//             message:"got all comments by users id"
+//         })
+//     }catch (err){
+//         console.log(err)
+//         res.json({
+//             error:err
+//         })
+//     }
+// })
 
     // get comments by video id 
     router.get('/video/:video_id', async (req, res) => {
+        const {video_id} = req.params
         try {
-            let commentByVideoID = await db.any(`SELECT * FROM comments WHERE video_id = '${req.params.video_id}' `)
+           const commentByVideoID = await comment.getCommentsByVideoID(video_id)
             res.json({
                 payload: commentByVideoID,
-                message:"got all comments by video id"
+                message:"got all comments by video id",
+                err: false
             })
         }catch (err){
-            console.log(err)
-            res.json({
-                error:err
+            console.log(error)
+            res.status(500).json({
+                payload: null,
+                msg: error,
+                err: true
             })
         }
     });
 
         //post a new comment
 router.post('/video/:video_id', async (req, res) => {
+    const {users_id, comment_body} = req.body
+    console.log(req.body)
     try {
-        await db.none(`INSERT INTO comments(users_id, comment_body) VALUES(${req.body.users_id}, ${req.body.comment_body})`)
-        res.json({ message: "comment added"})
+        const addComment = await comment.addnewComment(users_id, comment_body)
+        res.json({
+            payload: addComment,
+            msg: "comment added",
+            error: false
+        })
     }catch(err) {
         console.log("err", err)
-        res.json({error: err})
+        console.log(error)
+        res.status(500).json({
+            payload: null,
+            msg: error,
+            err: true
+        })
     }
 });
 
 //Delete comment
 router.delete('/', async (req, res) => {
+    const {comment_id} = req.params
     console.log(req.body);
     try {
-      let comment_id = req.body.comment_id;
-      let comment_body = req.body.comment_body;
-  
-      let insertQuery = `
-        DELETE FROM comments
-        WHERE comment_id= $1 AND comment_body = $2`;
-      await db.none(insertQuery, [comment_id, comment_body]);
-      console.log(req.body);
+      const deleteComm = await comment.deleteCommentByID(comment_id)
```

Comment:
This is fantastic. Move the DB stuff to the DAO (data access object) layer. No way did those queries belong in the routes. Good job.